File: nextGreaterI.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> nextGreaterElement(vector<int> &nums1, vector<int> &nums2) {
  vector<int> res;
  // sort(nums2.begin(),nums2.end());
  for (int i = 0; i < nums1.size(); i++) {
    auto it = find(nums2.begin(), nums2.end(), nums1[i]);
    auto j = it;
    for (j = it; j != nums2.end(); j++) {
      if (*j > nums1[i]) {
        res.push_back(*j);
        break;
      }
    }
    if (j == nums2.end()) {
      res.push_back(-1);
    }
  }
  return res;
}

int main() {
  vector<int> vec1 = {4, 1, 2};
  vector<int> vec2 = {1, 3, 4, 2};
  vector<int> ans = nextGreaterElement(vec1, vec2);
  for (auto i : ans)
    cout << i << " ";
}

/*
[4,1,2], [1,3,4,2] = [-1,3,-1]
[2,4], [1,2,3,4] = [3, -1]
*/

File: nextGreaterI.js
var nextGreaterElement = function (nums1, nums2) {
	const ans = [];
	for (let i = 0; i < nums1.length; i++) {
		let index = nums2.indexOf(nums1[i]);
		let temp = 0;
		for (let j = index; j < nums2.length; j++) {
			if (nums2[j] > nums1[i]) {
				ans.push(nums2[j]);
				temp = nums2[j];
				break;
			}
		}
		if (temp === 0) ans.push(-1);
	}
	return ans;
};

const arr1 = [4, 1, 2];
const arr2 = [1, 3, 4, 2];
console.log(nextGreaterElement(arr1, arr2));

/*
[4,1,2], [1,3,4,2] = [-1,3,-1]
[2,4], [1,2,3,4] = [3, -1]
*/
